Sources/BitArrayModule/BitSet/BitSetSetAlgebra.swift

```diff
@@ -0,0 +1,123 @@
+  }
+  
+  internal mutating func _forceInsert(_ newMember: Int) {
+    precondition(newMember >= 0, "Inserts can only be executed with a positive number. Bit sets do not hold negative values.")
+    while (storage.count-1 < newMember) {
+      storage.storage.append(0)
+    }
+    storage[newMember] = true
+  }
+  
+  @discardableResult
+  public mutating func remove(_ member: Int) -> Bool {
+    precondition(member >= 0, "Removals can only be executed with a positive number. Bit sets do not hold negative values.")
+    if (member >= storage.endIndex) {
+      return false
+    }
+    
+    let returnVal = storage[member]
+    storage[member] = false
+    return returnVal
+  }
+  
+  public func union(_ other: BitSet) -> BitSet {
+    var newBitSet = self
+    newBitSet.formUnion(other)
+    return newBitSet
+  }
+  
+  public func intersection(_ other: BitSet) -> BitSet {
+    var newBitSet = self
+    newBitSet.formIntersection(other)
+    return newBitSet
+  }
+  
+  public func symmetricDifference(_ other: BitSet) -> BitSet {
+    var newBitSet = self
+    newBitSet.formSymmetricDifference(other)
+    return newBitSet
+  }
+  
+  
+  public mutating func formUnion(_ other: BitSet) {
+    
+    if (other.storage.count < self.storage.count) {
+      for i in 0..<other.storage.storage.count {
+        self.storage.storage[i] |= other.storage.storage[i]
+      }
+    } else if (self.storage.count < other.storage.count){
+      for j in 0..<self.storage.storage.count {
+        self.storage.storage[j] |= other.storage.storage[j]
+      }
+      for a in self.storage.count..<other.storage.count {
+        if(other.storage[a]) {
+          self._forceInsert(a)
+        }
+      }
+    } else {
+      for b in 0..<self.storage.storage.count {
+        self.storage.storage[b] |= other.storage.storage[b]
+      }
+    }
+    
+  }
+  
+  public mutating func formIntersection(_ other: BitSet) {
+    let size: Int = (self.storage.storage.count >= other.storage.storage.count) ? other.storage.storage.count : self.storage.storage.count // take the set with the smaller BitArray
+    
+    for i in 0..<size {
+      self.storage.storage[i] &= other.storage.storage[i]
+    }
+    
+    for i in size..<self.storage.storage.count {
+      self.storage.storage[i] = 0
+    }
+  }
+  
+  public mutating func formSymmetricDifference(_ other: BitSet) { // can be optimized later
+    
+    if (other.storage.storage.count > self.storage.storage.count) {
+      for i in 0..<self.storage.storage.count {
+        self.storage.storage[i] ^= other.storage.storage[i]
+      }
+      for i in self.storage.storage.count..<other.storage.storage.count {
+        self.storage.storage.append(other.storage.storage[i])
+      }
+    } else {
+      for i in 0..<other.storage.storage.count {
+        self.storage.storage[i] ^= other.storage.storage[i]
+      }
+    }
+  }
+  
+  public static func == (lhs: BitSet, rhs: BitSet) -> Bool {
+    return (lhs.storage == rhs.storage)
+  }
```

Comment:
It seems a bit odd to have an extension granting `Equatable` conformance, which contains so many other things besides just an implementation of the `==` operator.

Also, this implementation appears to be at best unintuitive, and perhaps outright incorrect, considering how element removal currently works:

```
let a: BitSet = []
var b = a
b.insert(100)
b.remove(100)
print(a == b)           // I would expect `true`, but I believe it is currently `false`
```

Reply:
Yes, you pointed that out earlier in a comment. It will be corrected soon and I will answer that comment soon along with other comments